### PR TITLE
refactor kafka sender

### PR DESCRIPTION
**Pull request by lispc**
1. split the old kafka sender class using channel, now a sender and a receiver
2. kafka messages are flushed regularly now
3. add kafka message check into integrated test

### CODE REVIEW

**Comment by HAOYUatHZ** [comment]
why not use "stable" but "1.49.0" specifically?

**Comment by HAOYUatHZ** [comment]
To me I may prefer 
renaming `XXXMessageReceiver` to MessageManager or something...    it's now a producer/sender and a consumer/receiver ATM

naming it with  Receiver  might not be accurate

**Comment by HAOYUatHZ** [comment]
well the branch protection rules aren't right.... I have just updated it.                              

It won't work this time but we can still force merging

File: src/message.rs [hunk 1]
```diff
@@ -146,50 +133,105 @@ impl KafkaMessageSender {
...
-                ticker.tick().await;
-                unsafe {
-                    (*self_ptr).flush();
+    pub fn start(self) {
+        let mut last_flush_time = Instant::now();
+        let flush_interval = std::time::Duration::from_millis(100);
+        let timeout_interval = std::time::Duration::from_millis(100);
+        loop {
+            if self.is_block() {
+                // skip receiving from channel, so main server can know something goes wrong
```
Comment (HAOYUatHZ) [inline src/message.rs:149]: maybe add logging here?  

if repeatedly logging "sleeping for XXX" we then can know something goes wrong

File: src/message.rs [hunk 2]
```diff
@@ -146,50 +133,105 @@ impl KafkaMessageSender {
...
+                // sleep to avoid cpu 100% usage
+                thread::sleep(flush_interval);
+            } else {
+                match self.receiver.recv_timeout(timeout_interval) {
+                    Ok((topic, message)) => {
```
Comment (HAOYUatHZ) [inline src/message.rs:154]: should we also add logging here?

**Comment by HAOYUatHZ** [comment]
I think I'm fine with this PR even without adding logging for now

**Review by HAOYUatHZ — APPROVED** [review]

File: src/message.rs (hunk 1 above)
Comment (lispc) [reply src/message.rs:149]: good idea. will add log later

File: src/message.rs (hunk 2 above)
Comment (lispc) [reply src/message.rs:154]: Ok(xx) and Timeout are both normal paths. So log here is not needed.

**Pull request merged**